### DISCUSSION TITLE
I am using jackson-databind depedency of version 2.14.2 but getting an error getType() not found.

### DISCUSSION THREAD

**Discussion by aayushRedHat** [FasterXML/jackson-databind#4213] [General]
I am getting the following error after executing Spring-Camel Application: 
Caused by: java.lang.NoSuchMethodError: com.fasterxml.jackson.databind.introspect.AnnotatedMember.getType(Lcom/fasterxml/jackson/databind/type/TypeBindings;)Lcom/fasterxml/jackson/databind/JavaType;

I checked the AnnotatedMember Class but there is not method for getType(). Can anyone suggest the appropriate version of the dependency containing the getType() method?

**Comment by JooHyukKim** [comment]
`AnnotatedMember` is an abstract class and does not implement `getType()` from its supser class  `Annotated`.

**Reply by JooHyukKim** [reply]
Sounds like dependency issue. You can check out https://github.com/FasterXML/jackson-databind main page or other sources on internet about it.

**Comment by cowtowncoder** [comment]
99% certainly incompatibility between Jackson (jackson-databind) version that is deployed (in classpath), vs. what code calling `AnnotatedMember.getType()` was compiled with.

This method came from `Annotated` base class, and as per:

https://www.javadoc.io/doc/com.fasterxml.jackson.core/jackson-databind/2.7.9/index.html

```
Deprecated. Since 2.7 Use [getType()](https://www.javadoc.io/static/com.fasterxml.jackson.core/jackson-databind/2.7.9/com/fasterxml/jackson/databind/introspect/Annotated.html#getType()) instead. To be removed from 2.8.
```

So something was compiled against Jackson 2.7 or earlier and for some reason used this method. Method hasn't been part of jackson-databind since maybe 2.8.
Whatever that something is hopefully has a newer version that does not use this method.

Note that this deprecation is unusual, Jackson project does not typically deprecate and remove methods of public API within minor releases. But if I remember correctly there was something special why this particular method could no longer be supported, when type resolution problems were fixed.

**Comment by teerthpatel0408** [comment]
@aayushRedHat  were you able to resolve the error as I am facing same problem when integrating swagger

**Reply by JooHyukKim** [reply]
Did you try what was suggested above?

**Reply by JooHyukKim** [reply]
Most likely that will do

**Reply by pjfanning** [reply]
I don't know what build tool you are using. Build tools like Maven and Gradle provide ways to list your dependencies and the versions for those dependencies. If you don't use a build tool and just dump lots of jar files in a directory, you can still check the file names of the jars (which typically include a version number). The versions of all the jackson jars should be made to match.

If you still get the issue then, can you provide us with the full NoSuchMethodError stack trace. This will tell us what is making the call to AnnotatedMember.getType.

**Reply by teerthpatel0408** [reply]
the below is the error message 

```
{
  "status": "FAIL",
  "error": [
    {
      "code": "500",
      "field": null,
      "description": "'com.fasterxml.jackson.databind.JavaType com.fasterxml.jackson.databind.introspect.AnnotatedMember.getType(com.fasterxml.jackson.databind.type.TypeBindings)'",
      "info": "java.lang.NoSuchMethodError: 'com.fasterxml.jackson.databind.JavaType com.fasterxml.jackson.databind.introspect.AnnotatedMember.getType(com.fasterxml.jackson.databind.type.TypeBindings)'\\n\\tat io.swagger.jackson.ModelResolver.resolve(ModelResolver.java:402)\\n\\tat io.swagger.jackson.ModelResolver.resolve(ModelResolver.java:189)\\n\\tat io.swagger.converter.ModelConverterContextImpl.resolve(ModelConverterContextImpl.java:100)\\n",
      "severity": "ERROR",
      "category": "REQUEST",
      "causes": [],
      "errorIdentifiers": {

      }
    }
  ]
}
```

when accessing the URL "http://localhost:8013/api/swagger.json"

also I have below dependencies 

```
               <dependency>
			<groupId>com.fasterxml.jackson.core</groupId>
			<artifactId>jackson-core</artifactId>
			<version>2.15.0</version>
		</dependency>

		<dependency>
			<groupId>com.fasterxml.jackson.core</groupId>
			<artifactId>jackson-databind</artifactId>
			<version>2.15.0</version>
		</dependency>

		<dependency>
			<groupId>com.fasterxml.jackson.core</groupId>
			<artifactId>jackson-annotations</artifactId>
		</dependency>
		
                <dependency>
			<groupId>com.fasterxml.jackson</groupId>
			<artifactId>jackson-bom</artifactId>
			<version>2.13.5</version>
			<type>pom</type>
		</dependency>

                <dependency>
			<groupId>io.swagger</groupId>
			<artifactId>swagger-jersey2-jaxrs</artifactId>
			<version>1.5.17</version>
		</dependency>
```

also I am using jax-rs annotations

**Reply by pjfanning** [reply]
* upgrade to a much newer version of swagger-jersey2-jaxrs
* https://mvnrepository.com/artifact/io.swagger/swagger-jersey2-jaxrs
* 1.6.15 is the latest release
* using 8 and 10 year old jars is a recipe for disaster and then trying to mix them with newer release of jackson - very bad idea

**Comment by pjfanning** [comment]
There is Stackoverflow question about this too.

https://stackoverflow.com/questions/77271263/java-lang-nosuchmethoderror-jackson-databind-introspect-annotatedmember-gettype

It has a stacktrace suggesting that that user was using an old swagger jar that was created to work with an old version of jackson. Upgrading the swagger jars should help.

**Reply by pjfanning** [reply]
Upgrading to the latest version of camel-jackson should help too as this seems to be built with a much newer version of jackson.

https://mvnrepository.com/artifact/org.apache.camel.springboot/camel-jackson-starter/4.10.0

**Reply by pjfanning** [reply]
https://issues.apache.org/jira/browse/CAMEL-15588 is the Camel fix to react to Jackson dropping AnnotatedMember.getType

**Reply by teerthpatel0408** [reply]
@pjfanning can you pls tell the particular steps which I need to follow so that I can resolve issue

**Reply by pjfanning** [reply]
* this code was removed in 2015 - why are you using 10 year jars and not using up to date releases? https://github.com/FasterXML/jackson-databind/commit/827f082f536b55b443b5c0ad999d142ac3699532
* looks like the code was removed due to https://github.com/FasterXML/jackson-databind/issues/76

**Reply by teerthpatel0408** [reply]
the thing is actually the code was implemented in 2017-18 around and now I was told to add swagger 
Is their any way to resolve the issue without disturbing the existing code ?

**Reply by pjfanning** [reply]
* old code means more bugs - you should upgrade
* if you insist on using ancient jars, downgrade to ancient jackson too - you can't use jars from different eras
* https://mvnrepository.com/artifact/io.swagger/swagger-core/1.5.17 appears to expect jackson 2.8.9

**Reply by pjfanning** [reply]
looking at the commit that removed this method - you probably need to go back to jackson-databind 2.6.7 to find a release that has the getType method.

**Reply by teerthpatel0408** [reply]
now when I downgraded jackson version i am getting response of http://localhost:8013/api/swagger.json but getting error when accessing http://localhost:8013/swagger-ui/index.html,basically the error is as below 

500 : {"status":"FAIL","error":[{"code":"500","field":null,"description":"HTTP 500 Internal Server Error","info":"javax.ws.rs.InternalServerErrorException: HTTP 500 Internal Server Error\\n\\tat org.glassfish.jersey.server.internal.MappableExceptionWrapperInterceptor.aroundWriteTo(MappableExceptionWrapperInterceptor.java:65)\\n\\tat org.glassfish.jersey.message.internal.WriterInterceptorExecutor.proceed(WriterInterceptorExecutor.java:139)\\n\\tat org.glassfish.jersey.message.internal.MessageBodyFactory.writeTo(MessageBodyFactory.java:1116)\\nCaused by: org.glassfish.jersey.message.internal.MessageBodyProviderNotFoundException: MessageBodyWriter not found for media type=application\\/yaml, type=class io.swagger.models.Swagger, genericType=class io.swagger.models.Swagger.\\n\\tat org.glassfish.jersey.message.internal.WriterInterceptorExecutor$TerminalWriterInterceptor.aroundWriteTo(WriterInterceptorExecutor.java:224)\\n\\tat org.glassfish.jersey.message.internal.WriterInterceptorExecutor.proceed(WriterInterceptorExecutor.java:139)\\n\\tat org.glassfish.jersey.server.internal.JsonWithPaddingInterceptor.aroundWriteTo(JsonWithPaddingInterceptor.java:85)\\n","severity":"ERROR","category":"REQUEST","causes":[],"errorIdentifiers":{}}]} http://localhost:8013/api/swagger.yaml

It would be thankfull if anyone could help me

**Reply by cowtowncoder** [reply]
@teerthpatel0408  I would suggest that at this point you do try to use a newer version of `swagger-jersey2-jaxrs` (or whatever depends on Jackson). Your stack trace has no info on Jackson issue and I don't think anyone here has much interest in software archeology of such old versions.

**Reply by teerthpatel0408** [reply]
ok will try 

Also I am getting white label error page 404 not found when integrating swagger openapi with springboot 3 in which I use spring annotations . also when I checked my endpoints are running in postman but when trying to access http://localhost:8012/swagger-ui/index.html getting whitelabel error
